### PR TITLE
Add support for Vector Search Indexes

### PR DESCRIPTION
**Pull request by alexott**
## Changes

This is initial work - right now things are blocked by errors in OpenAPI spec, and as result, Go SDK has incorrect structs. So we need to wait until spec is fixed

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

**Comment by alexott** [comment]
Regarding integration tests - it's not clear how to programmatically pull models from the Marketplace :-(

**Comment by alexott** [comment]
@mgyucht to add acceptance tests we need to fix a number of issues: #3388, #3383, #3375, plus solve the problem with registration of models from the marketplace...

File: docs/resources/vector_search_index.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  * `indexed_row_count` - Number of rows indexed
+  * `ready` - Whether the index is ready for search
+  * `index_url` - Index API Url to be used to perform operations on the index
+
+## Import
+
+The resource can be imported using the name of the Vector Search Endpoint
+
+```bash
+terraform import databricks_vector_search_endpoint.this <endpoint-name>
```
Comment (mgyucht) [inline docs/resources/vector_search_index.md:78]: This needs to be updated to refer to index?

File: vectorsearch/resource_vector_search_index.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+			common.CustomizeSchemaPath(s, "primary_key").SetRequired()
+			common.CustomizeSchemaPath(s, "status").SetReadOnly()
+			common.CustomizeSchemaPath(s, "creator").SetReadOnly()
+			common.CustomizeSchemaPath(s, "name").SetRequired()
+			common.CustomizeSchemaPath(s, "index_type").SetRequired()
+			common.CustomizeSchemaPath(s, "delta_sync_index_spec", "pipeline_id").SetReadOnly()
+			return s
+		})
+
+	return common.Resource{
```
Comment (mgyucht) [inline vectorsearch/resource_vector_search_index.go:66]: No update API (yet)?

**Review by mgyucht — APPROVED** [review]
LGTM with two small nits, discussed adding integration test in a follow-up offline.

File: vectorsearch/resource_vector_search_index.go (hunk 2 above)
Comment (alexott) [reply vectorsearch/resource_vector_search_index.go:66]: No API - the same as for VS Endpoints and Online Tables

**Pull request merged**